Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/CwaDialog.kt

```diff
@@ -0,0 +1,201 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.RawValue
+import kotlin.experimental.ExperimentalTypeInference
+
+@DslMarker
+annotation class CwaDialogDsl
+
+typealias DialogAction = () -> Unit
+
+data class CwaDialog(
+    val texts: CwaDialogTexts,
+    val actions: CwaDialogActions,
+    val options: CwaDialogOptions
+)
+
+@CwaDialogDsl
+class CwaDialogBuilder {
+    private var config = CwaDialogTexts()
+    private var actions = CwaDialogActions()
+    private var options = CwaDialogOptions()
+
+    fun configureTexts(lambda: CwaDialogTextsBuilder.() -> Unit) {
+        config = CwaDialogTextsBuilder().apply(lambda).build()
+    }
+
+    fun configureActions(lambda: CwaDialogActionsBuilder.() -> Unit) {
+        actions = CwaDialogActionsBuilder().apply(lambda).build()
+    }
+
+    fun configureOptions(lambda: CwaDialogOptionsBuilder.() -> Unit) {
+        options = CwaDialogOptionsBuilder().apply(lambda).build()
+    }
+
+    fun build() = CwaDialog(config, actions, options)
+}
+
+@Parcelize
+data class CwaDialogTexts(
+    val title: @RawValue IntOrString? = null,
+    val message: @RawValue IntOrString? = null,
+    val positiveButton: @RawValue IntOrString? = null,
+    val negativeButton: @RawValue IntOrString? = null,
+    val neutralButton: @RawValue IntOrString? = null,
+) : Parcelable
+
+@CwaDialogDsl
+class CwaDialogTextsBuilder {
+    private var title = IntOrString()
+    private var message = IntOrString()
+    private var positiveButton = IntOrString()
+    private var negativeButton = IntOrString()
+    private var neutralButton = IntOrString()
+
+    @OptIn(ExperimentalTypeInference::class)
```

Comment:
Unfotunately you can't overload functions with lambda params of different types in Kotlin by default. I had to opt to use this annotation. https://youtrack.jetbrains.com/issue/KT-11265/Overload-resolution-by-lambda-return-type#focus=Comments-27-4303340.0-0

I understand if you don't want to include experimental features in. I can make the DSL usable without it, but it won't look as nice and tidy :/

Reply:
Removed it.